cli/compose/convert/service.go

```diff
@@ -36,9 +37,19 @@ func Services(
 		if err != nil {
 			return nil, errors.Wrapf(err, "service %s", service.Name)
 		}
-		configs, err := convertServiceConfigObjs(client, namespace, service.Configs, config.Configs)
-		if err != nil {
-			return nil, errors.Wrapf(err, "service %s", service.Name)
+		var configs []*swarm.ConfigReference
+		if len(service.Configs) > 0 {
```

Comment:
@dnephin secrets support were already there in swarm mode when we added them in compose v3. If we point to a 1.12 daemon (that may not have secret ?), the bug will be there for secret too yep.